src/components/graph/Graph.jsx

```diff
@@ -237,6 +237,26 @@ export default class Graph extends React.Component {
             draggedNode.x += d3Event.dx;
             draggedNode.y += d3Event.dy;
 
+            // limit the node inside canvas boundaries
+            let { width, height } = this.state.config;
+            const adjustSize = draggedNode.size ? draggedNode.size / 2 : 0;
+            const left = adjustSize;
```

Comment:
The range is fixed for a canvas, we can store it as global variable.

Reply:
we need to adjust the limit by node size, so they are not fixed